### PR TITLE
Media refactor

### PR DESCRIPTION
**Pull request by NejcZdovc**

### CODE REVIEW

File: include/bat/ledger/ledger_client.h [hunk 1]
```diff
@@ -30,7 +30,7 @@ LEDGER_EXPORT enum URL_METHOD {
 using PublisherInfoCallback = std::function<void(Result,
     std::unique_ptr<PublisherInfo>)>;
 using MediaPublisherInfoCallback = std::function<void(Result,
-    std::unique_ptr<MediaPublisherInfo>)>;
+    std::unique_ptr<PublisherInfo>)>;
```
Comment (SergeyZhukovsky) [inline include/bat/ledger/ledger_client.h]: We can probably use the PublisherInfoCallback as it is the same

File: src/bat_publishers.cc [hunk 2]
```diff
@@ -59,7 +59,7 @@ double BatPublishers::concaveScore(const uint64_t& duration) {
   return (std::sqrt(b2_ + a4_ * duration) - b_) / (double)a2_;
 }
 
-std::string getProviderName(const ledger::PublisherInfo::id_type publisher_id) {
+std::string getProviderName(const std::string publisher_id) {
```
Comment (SergeyZhukovsky) [inline src/bat_publishers.cc]: const std::string&

File: src/bat_publishers.cc [hunk 3]
```diff
@@ -69,7 +69,7 @@ std::string getProviderName(const ledger::PublisherInfo::id_type publisher_id) {
   return "";
 }
 
-bool ignoreMinTime(const ledger::PublisherInfo::id_type publisher_id) {
+bool ignoreMinTime(const std::string publisher_id) {
```
Comment (SergeyZhukovsky) [inline src/bat_publishers.cc]: const std::string&

File: src/bat_get_media.cc [hunk 4]
```diff
@@ -119,15 +119,22 @@ void BatGetMedia::getPublisherInfoDataCallback(const std::string& mediaId, const
...
+      std::map<std::string, ledger::TwitchEventInfo>::const_iterator iter = twitchEvents.find(media_key);
+      if (iter != twitchEvents.end()) {
+        oldEvent = iter->second;
+      }
+
+      ledger::TwitchEventInfo newEvent(twitchEventInfo);
+      newEvent.status_ = getTwitchStatus(oldEvent, newEvent);
+
+      uint64_t realDuration = getTwitchDuration(oldEvent, newEvent);
+      twitchEvents.emplace(media_key, newEvent);
```
Comment (SergeyZhukovsky) [inline src/bat_get_media.cc]: emplace doesn't replace the existing element, it just inserts new if the key doesn't exist, you never update the event value with that code, it's better to do `twitchEvents[media_key] = newEvent;`

File: src/bat_get_media.cc [hunk 5]
```diff
@@ -148,13 +155,19 @@ void BatGetMedia::getPublisherInfoDataCallback(const std::string& mediaId, const
...
```
Comment (SergeyZhukovsky) [inline src/bat_get_media.cc]: the same as above

File: include/bat/ledger/ledger_client.h (hunk 1 above)
Comment (NejcZdovc) [reply include/bat/ledger/ledger_client.h]: done

File: src/bat_publishers.cc (hunk 2 above)
Comment (NejcZdovc) [reply src/bat_publishers.cc]: done

File: src/bat_publishers.cc (hunk 3 above)
Comment (NejcZdovc) [reply src/bat_publishers.cc]: done

File: src/bat_get_media.cc (hunk 4 above)
Comment (NejcZdovc) [reply src/bat_get_media.cc]: done

File: src/bat_get_media.cc (hunk 5 above)
Comment (NejcZdovc) [reply src/bat_get_media.cc]: done

**Pull request merged**